Sources/UtiliKit/ActiveLabel/ActiveLabel.swift

```diff
@@ -0,0 +1,324 @@
+        // The offset is enough to make sure that the gradient doesn't originally show on screen.
+        return bounds.width * 0.4
+    }
+    
+    /**
+     When text is set to `nil` the loading views will display. When text is non-`nil` the loading views will be hidden.
+     */
+    override var text: String? {
+        didSet {
+            textDidUpdate()
+        }
+    }
+    
+    override var isHidden: Bool {
+        didSet {
+            if isHidden {
+                hideLoadingViews()
+            } else if !isHidden && state == .loading {
+                showLoadingViews()
+            }
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        commonInit()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        commonInit()
+    }
+    
+    convenience init(frame: CGRect, configuration: ActiveLabel.ActiveLabelConfiguration) {
+        self.init(frame: frame)
+        
+        self.estimatedNumberOfLines = configuration.estimatedNumberOfLines
+        self.finalLineTrailingInset = configuration.finalLineTrailingInset
+        self.finalLineLength = configuration.finalLineTrailingInset
+        self.loadingViewColor = configuration.loadingViewColor
+        self.loadingLineHeight = configuration.loadingLineHeight
+        self.loadingLineVerticalSpacing = configuration.loadingLineVerticalSpacing
+        self.loadingAnimationDuration = configuration.loadingAnimationDuration
+        self.loadingAnimationDelay = configuration.loadingAnimationDelay
+        
+        configurationChanged()
+    }
+    
+    private func commonInit() {
+        textDidUpdate()
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        loadingViews.forEach { [weak self] (view) in
+            guard let self = self else { return }
+            if let maskLayer = view.layer.mask {
+                maskLayer.bounds = CGRect(x: 0, y: 0, width: self.bounds.width + (maskOffset * 2), height: self.bounds.height)
+                maskLayer.frame = CGRect(x: -maskOffset, y: 0, width: self.bounds.width + (maskOffset * 2), height: self.bounds.height)
+            }
+        }
+    }
+    
+    override func prepareForInterfaceBuilder() {
+        super.prepareForInterfaceBuilder()
+        
+        isDisplayingInStoryboard = true
+        state = showLoadingViewsInStoryboard ? .loading : .text("")
+    }
+    
+    /**
+     You must call this function to reset the loading views after you have changed a configuration value.
+     */
+    func configurationChanged() {
+        loadingViews.forEach { $0.removeFromSuperview() }
+        loadingViews.removeAll()
+        
+        if state == .loading {
+            showLoadingViews()
+        }
+    }
+}
+
+private extension ActiveLabel {
+    func textDidUpdate() {
+        state = text == nil ? .loading : .text(text ?? "")
```

Comment:
Very minor, but a "Swiftier" way to do this might be to add a custom initializer on `ActiveLabel.State`. Also, I think we might be able to accomplish it without using the nil-coalescing operator.
```swift
enum State: Equatable {
    case loading
    case text(String)

    init(text: String?) {
        guard let text = text else { self = .loading }
        self = .text(text)
    }
}
```

Reply:
Fixed.